r/R/util.R

```diff
@@ -251,3 +251,25 @@ augment_io_error_msg <- function(e, call, schema = NULL, format = NULL) {
   handle_augmented_field_misuse(msg, call)
   abort(msg, call = call)
 }
+
+simulate_data_frame <- function(schema) {
+
+  arrays <- lapply(
+    schema$fields,
+    function(field) tryCatch(
+      concat_arrays(type = field$type),
+      error = function(...) concat_arrays(type = NULL)
+    )
+  )
+
+  vectors <- lapply(
+    arrays,
+    function(array) tryCatch(
+      as.vector(array),
+      error = function(...) vctrs::unspecified()
+    )
+  )
+
+  names(vectors) <- names(schema)
+  tibble::new_tibble(vectors, nrow = 0)
```

Comment:
```suggestion
  vctrs::new_data_frame(vectors, nrow = 0)
```

and we should import that not :: it